### PR TITLE
Export test metadata feature

### PR DESCRIPTION
**Pull request by alroniks**
It is the implementation feature pull request, based on [top of refactoring](https://github.com/oat-sa/extension-tao-testqti/pull/2326) the whole export for tests in the extension.

Task is [REL-880](https://oat-sa.atlassian.net/browse/REL-880)

### CODE REVIEW

**Review by mianchenia — APPROVED** [review]
LGTM!

**Review by yaraslau-kavaliou — APPROVED** [review]
- [x]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful
- [x]  Pull request's target is not master

File: models/classes/export/AbstractTestExport.php [hunk 1]
```diff
@@ -46,30 +45,23 @@
 abstract class AbstractTestExport implements ExportHandler, PhpSerializable
 {
     use PhpSerializeStateless;
-    use EventManagerAwareTrait;
+    use EventManagerAwareTrait; // todo: call firing event via method to have abilty to override
```
Comment (augustas) [inline models/classes/export/AbstractTestExport.php:48]: To override for what purpose? Do you have a use case in mind?

File: models/classes/export/AbstractTestExport.php [hunk 2]
```diff
@@ -151,6 +141,16 @@ public function export($formValues, $destination): Report
         return $report;
     }
 
+    protected function fireTestExportEvent(Resource $test)
```
Comment (augustas) [inline models/classes/export/AbstractTestExport.php:144]: `trigger` is a more expected term, but `fire` works as well :)

**Review by augustas — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: models/classes/export/AbstractTestExport.php (hunk 1 above)
Comment (alroniks) [reply models/classes/export/AbstractTestExport.php:48]: I've already done it, but I forgot to remove the comment.
https://github.com/oat-sa/extension-tao-testqti/pull/2329/files#diff-1fc574c1b6d9f7e863bd7ffd50bb016655b86e5f60328619245741da86ab2d5bR135

File: models/classes/export/AbstractTestExport.php (hunk 2 above)
Comment (alroniks) [reply models/classes/export/AbstractTestExport.php]: Renamed

**Pull request merged**